js/bcex.js

```diff
@@ -297,30 +495,36 @@ module.exports = class bcex extends Exchange {
     }
 
     async fetchOrder (id, symbol = undefined, params = {}) {
+        if (typeof symbol === 'undefined')
+            throw new ExchangeError (this.id + ' fetchOrder requires a symbol argument');
         await this.loadMarkets ();
         let request = {
             'symbol': this.marketId (symbol),
             'trust_id': id,
         };
         let response = await this.privatePostApiOrderOrderInfo (this.extend (request, params));
         let order = response['data'];
-        let timestamp = order['created'] * 1000;
+        let timestamp = this.safeInteger (order, 'created') * 1000;
         let status = this.parseOrderStatus (order['status']);
+        let side = this.safeString (order, 'flag');
+        if (side === 'sale')
+            side = 'sell';
+        // Can't use parseOrder because the data format is different btw endpoint for fetchOrder and fetchOrders
```

Comment:
We design the implementations of parsing functions to handle the differences. Rhe same is true for all parsing methods and all exchanges:

- `parseOrder` will parse the results of `fetchOrder`, `fetchOpenOrders`, `fetchClosedOrders`, `fetchOrders`
- `parseTrade` will parse both public `fetchTrades` and private `fetchMytrades`
- `parseTransaction` will parse `fetchDeposits`, `fetchWithdrawals` and `fetchTransactions`
- `parseTicker` will parse `fetchTicker` and `fetchTickers`
- ...

Thus, parser methods are the actual unification workhorses.